src/Microsoft.VisualBasic/tests/VB/SpecialDirectoriesTests.vb

```diff
@@ -0,0 +1,99 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+Option Explicit On
+Option Strict On
+
+Imports System.Environment
+Imports Microsoft.VisualBasic.FileIO
+Imports Xunit
+Namespace Microsoft.VisualBasic.Tests.VB
+    Public NotInheritable Class SpecialDirectoriesTests
+        Private Shared ReadOnly Separators() As Char = {
+            IO.Path.DirectorySeparatorChar,
+            IO.Path.AltDirectorySeparatorChar
+            }
+
+        <Fact>
+        Public Shared Sub AllUsersApplicationDataFolderTest()
+            Assert.Throws(Of PlatformNotSupportedException)(Function() SpecialDirectories.AllUsersApplicationData)
+        End Sub
+
+        <Fact>
+        Public Shared Sub CurrentUserApplicationDataFolderTest()
+            Assert.Throws(Of PlatformNotSupportedException)(Function() SpecialDirectories.CurrentUserApplicationData)
+        End Sub
+
+        <Fact>
+        Public Shared Sub DesktopFolderTest()
+            Dim Env_Desktop As String = Environment.GetFolderPath(SpecialFolder.Desktop).Trim.TrimEnd(Separators).TrimEnd
+            If Env_Desktop.Length = 0 Then
+                Assert.Throws(Of IO.DirectoryNotFoundException)(Function() SpecialDirectories.Desktop)
+            Else
+                Dim FileIO_Desktop As String = SpecialDirectories.Desktop.TrimEnd(Separators)
+                Assert.True(Env_Desktop = FileIO_Desktop, $"{Env_Desktop} <> {FileIO_Desktop}")
+            End If
+        End Sub
+
+        <Fact>
+        Public Shared Sub MyDocumentsFolderTest()
+            If PlatformDetection.IsWindowsNanoServer Then
+                Exit Sub
+            End If
```

Comment:
Is this necessary? Is `MyDocuments` handled differently than other `SpecialFolder` cases?

Reply:
Yes, Nano does not support MyDocuments it is a known bug.

Reply:
It throws an exception but that code is supposed to be fixed when someone agrees what the fix should be.